feat: Docs for Java client (#945)
questdb/questdb.io: docs/reference/clients/java_ilp.md

```xml
<dependency>
    <groupId>org.questdb</groupId>
    <artifactId>questdb</artifactId>
    <version>6.4.3</version>
</dependency>
```
</TabItem>

<TabItem value="gradle">

```text
compile group: 'org.questdb', name: 'questdb', version: '6.4.3'
```
</TabItem>

</Tabs>
The code bellow creates an instance of a client, configures it to connect to a QuestDB server running on localhost
on a TCP port 9009. Then it sends two rows, each with one long column and one string column. It instructs the QuestDB 
server to assign a timestamp by using a local wall-clock. 

```java
package com.example;

import io.questdb.client.Sender;

public class SenderExample {
    public static void main(String[] args) {
        try (Sender sender = Sender.builder().address("localhost:9009").build()) {
            sender.table("inventors")
                    .longColumn("id", 0)
                    .stringColumn("name", "Nicola Tesla")
                    .atNow();
            sender.table("inventors")
                    .longColumn("id", 1)
                    .stringColumn("name", "Thomas Alva Edison")
                    .atNow();
        }
    }
}
```

## Example with TLS and Authentication enabled

> This sample configures a client to use TLS encryption for a connection to a QuestDB server. It will also instruct
> the client to authenticate.  

Reviewer: ```suggestion
This sample configures a client to use TLS encryption for a connection to a QuestDB server. It also instructs
the client to authenticate.  
```

```java
package com.example;

import io.questdb.client.Sender;

public class SenderExample {
    public static void main(String[] args) {
        try (Sender sender = Sender.builder()
                .address("clever-black-363-c1213c97.ilp.b04c.questdb.net:32074")
                .enableTls()
                .enableAuth("admin").authToken("GwBXoGG5c6NoUTLXnzMxw_uNiVa8PKobzx5EiuylMW0")
                .build()) {
            sender.table("inventors")
                    .longColumn("id", 0)
                    .stringColumn("name", "Nicola Tesla")
                    .atNow();
            sender.table("inventors")
                    .longColumn("id", 1)
                    .stringColumn("name", "Thomas Alva Edison")
                    .atNow();
        }
    }
}
```

> This configuration also works with [QuestDB Cloud](https://questdb.io/cloud/). If you are using a self-managed QuestDB

Reviewer: Should we hide this for later when Cloud is launch?

Reviewer: Let's stick to the current version then :)

with [authentication enabled](/docs/reference/api/ilp/authenticate) then `authToken` is the `d` portion of a JWK. 

> ## General Usage Pattern

Reviewer: Sentense case:
```suggestion
## General usage pattern
```

> 1. Create a client instance via `Sender.builder()`

Reviewer: Here and the rest for the numbered list: I'd add full stops at the end of a complete sentence.

Author: that's one thing I wanted to ask you. I'm never sure whether full stops should be at the end of an item or not. so now I know. I will know it. at least some time. thank you!

2. Use `table(CharSequence)` to select a table
3. Use `symbol(CharSequence, CharSequence)` to add all symbols. You must add symbols before adding other columns

> 4. Use `stringColumn(CharSequence, CharSequence)`, `longColumn(CharSequence, long)`,
>  `doubleColumn(CharSequence, double)`, `boolColumn(CharSequence, boolean)` and `timestampColumn(CharSequence, long)` to
> add all remaining columns

Reviewer: I'd consider use a list.
```suggestion
4. Use the following options to add all the remaining columns:
    * `stringColumn(CharSequence, CharSequence)`
    * `longColumn(CharSequence, long)`
     * `doubleColumn(CharSequence, double)`
     * `boolColumn(CharSequence, boolean)`
     * `timestampColumn(CharSequence, long)`

```

> 5. Use `#t(long)` to finish a row with an explicit timestamp. Alternatively, you can use `atNow()` which will add

Reviewer: I'd use present tense. 
```suggestion
5. Use `#t(long)` to finish a row with an explicit timestamp. Alternatively, you can use `atNow()` which adds
```

a timestamp on a server
6. Optionally: You can use `flush()` to send locally buffered data into a server

> 7. Go to the step no. 2 to start a new row

Reviewer: Step two is about selecting a table? I am not sure I can follow the instructions here...

8. Use `close()` to dispose the Sender after you no longer need it

> ## Designated Timestamp Considerations

Reviewer: ```suggestion
## Designated timestamp considerations
```

I think sentence case is the norm in our website.

> Client can either assign a [designated timestamp](/docs/concept/designated-timestamp/) explicitly or it can instruct
> a QuestDB server to automatically assign a timestamp upon receiving a row from a client. The code samples above use the latter
> options: `atNow()` makes the server to assign a timestamp by using a local wall-clock. When you need to assign
> a specific timestamp then use `at(long)` instead of `atNow()`. 

Reviewer: ```suggestion
To assign a timestamp to a row, a client has the following options:

* Assign a [designated timestamp](/docs/concept/designated-timestamp/) explicitly
* Instruct a QuestDB server to automatically assign a timestamp upon receiving a row from the client. 

The code samples above uses QuestDB to assign timestamp automatically: `atNow()` instructs the server to assign a timestamp by using a local wall-clock. 
To assign a specific timestamp, use `at(long)` instead of `atNow()`. 
```

:::note

QuestDB works best when data are ingested in a chronological order. This means rows with older timestamps are ingested
before rows with newer timestamps. Receiving out-of-order data can have a performance impact.

> See [this guide](/docs/guides/out-of-order-commit-lag) for more informations

Reviewer: ```suggestion
See [Out-of-order commit lag](/docs/guides/out-of-order-commit-lag) for more informations.
```

:::

> Bear in mind the client will not interpret the timestamp in any way. It will merely pass the timestamp
> to a server, and it's up to a server to interpret the timestamp. The default behaviour of a QuestDB server is to treat
> the timestamp as a number of nanoseconds since 1st Jan 1970 UTC, see [documentation](/docs/develop/insert-data/#timestamps)
> for more details.   

Reviewer: ```suggestion
Please note that he client does not interpret the timestamp in any way. It merely passes the timestamp
to a server, and it is up to a server to interpret the timestamp. 
The default behavior of a QuestDB server is to treat the timestamp as a number of nanoseconds since 1st Jan 1970 UTC. See [Timestamps](/docs/develop/insert-data/#timestamps) for more details.   
```

:::caution

> Default behaviour of QuestDB is to treat a designated timestamp as nanoseconds, but this does not mean you could use
> `System.nanoTime()` to get the current time and pass it to `at(long`). `System.nanoTime()` is only useful for measuring
> elapsed time and it is not related to any notion of system or wall-clock time.   
> 

Reviewer: ```suggestion

By default, QuestDB's engine treats a designated timestamp as nanoseconds, but this does not mean that `System.nanoTime()`  can be used to get the current time and pass it to `at(long)`. 
`System.nanoTime()` is only useful for measuring elapsed time,  and it is not related to any notion of system or wall-clock time.   

```

:::

## Other Considerations

> - Sender objects are not thread safe. If you need multiple threads to send data to QuestDB then each thread should have
> its own Sender instance. You can also use an object pool to re-use Senders.

Reviewer: ```suggestion
- Sender objects are not thread-safe. For multiple threads to send data to QuestDB, each thread should have
its own Sender instance. An object pool can also be used to re-use Senders.
```

> - Sender objects have to be closed after they are no longer use. Sender implements the `java.lang.AutoCloseable` interface
> thus you can use the try-with-resource pattern to make sure the Sender is closed. 

Reviewer: ```suggestion
- The Sender objects have to be closed after they are no longer in use. The Sender implements the `java.lang.AutoCloseable` interface, and therefore the try-with-resource pattern can be used to ensure that the Sender is closed. 
```

> - Client internally buffers row data and sends them to a server in batches when the buffers fill up. This improves
> performance significantly, but it also means that during a period of quiescence a client may keep previously written
> rows in the buffer and sends to a server only when additional rows are inserted and the buffer fill up. If this is
> undeseriable then you can call `flush()` to force sending internal buffers to a server even when the buffers are not
> full yet.

Reviewer: The first part (Client internally buffers row data and sends them to a server in batches when the buffers fill up. ): is it a default setting or if a user configures this way?I am guess the latter.
Also there are more than one buffers? ("when the buffers fill up).
```suggestion
- A client can be configured to buffer row data internally and sends them to a server in batches when the buffers are full.  This configuration improves
performance significantly, but it also means that during a period of quiescence, some written
rows may be retained in the buffers and are not sent  to a server until the buffers are full [TBD buffer or buffers]. The function `flush()` can be called to force sending internal buffers to a server, even when the buffers are not full yet.
```

To assign a specific timestamp, use `at(long)` instead of `atNow()`. 

:::note

QuestDB works best when rows are ingested in a chronological order. This means rows with older timestamps are ingested
before rows with newer timestamps. Receiving out-of-order data can have a performance impact.
See [Out-of-order commit lag](/docs/guides/out-of-order-commit-lag) for more informations.

:::

Please note that the client does not interpret the timestamp in any way. It merely passes the timestamp
to a server, and it is up to a server to interpret the timestamp. 
The default behavior of a QuestDB server is to treat the designated timestamp as a number of nanoseconds since 1st Jan 1970 UTC. See [Timestamps](/docs/develop/insert-data/#timestamps) for more details.   

:::caution

By default, QuestDB's engine treats a designated timestamp as nanoseconds, but this does not mean that `System.nanoTime()`  can be used to get the current time and pass it to `at(long)`. 
`System.nanoTime()` is only useful for measuring elapsed time,  and it is not related to any notion of system or wall-clock time.   

:::

> ## Other Considerations

Reviewer: ```suggestion
## Other considerations
```
Sorry I missed this.
